scripts/convert-bson-corpus-tests.php

```diff
@@ -151,34 +151,62 @@ function getParamsForValid(array $test, array $case)
     $code .= 'echo bin2hex(fromPHP(toPHP($canonicalBson))), "\n";' . "\n";
     $expect .= $expectedCanonicalBson . "\n";
 
+    $code .= "\n// Canonical BSON -> Native -> Canonical BSON\n";
+    $code .= 'echo bin2hex((string) MongoDB\BSON\Document::fromBSONString($canonicalBson)), "\n";' . "\n";
+    $expect .= $expectedCanonicalBson . "\n";
+
     $code .= "\n// Canonical BSON -> Canonical extJSON\n";
-    $code .= 'echo json_canonicalize(toCanonicalExtendedJSON($canonicalBson)), "\n";' . "\n";;
+    $code .= 'echo json_canonicalize(toCanonicalExtendedJSON($canonicalBson)), "\n";' . "\n";
+    $expect .= $expectedCanonicalExtJson . "\n";
+
+    $code .= "\n// Canonical BSON -> BSON object -> Canonical extJSON\n";
+    $code .= 'echo json_canonicalize(MongoDB\BSON\Document::fromBSONString($canonicalBson)->toCanonicalExtendedJSON()), "\n";' . "\n";
     $expect .= $expectedCanonicalExtJson . "\n";
 
     if (isset($relaxedExtJson)) {
         $code .= "\n// Canonical BSON -> Relaxed extJSON\n";
-        $code .= 'echo json_canonicalize(toRelaxedExtendedJSON($canonicalBson)), "\n";' . "\n";;
+        $code .= 'echo json_canonicalize(toRelaxedExtendedJSON($canonicalBson)), "\n";' . "\n";
+        $expect .= $expectedRelaxedExtJson . "\n";
+
+        $code .= "\n// Canonical BSON -> BSON object -> Relaxed extJSON\n";
+        $code .= 'echo json_canonicalize(MongoDB\BSON\Document::fromBSONString($canonicalBson)->toRelaxedExtendedJSON()), "\n";' . "\n";
         $expect .= $expectedRelaxedExtJson . "\n";
     }
 
     if (!$lossy) {
         $code .= "\n// Canonical extJSON -> Canonical BSON\n";
         $code .= 'echo bin2hex(fromJSON($canonicalExtJson)), "\n";' . "\n";
         $expect .= $expectedCanonicalBson . "\n";
+
+        $code .= "\n// Canonical extJSON -> BSON object -> Canonical BSON\n";
+        $code .= 'echo bin2hex((string) MongoDB\BSON\Document::fromJSON($canonicalExtJson)), "\n";' . "\n";
+        $expect .= $expectedCanonicalBson . "\n";
     }
 
     if (isset($degenerateBson)) {
         $code .= "\n// Degenerate BSON -> Native -> Canonical BSON\n";
         $code .= 'echo bin2hex(fromPHP(toPHP($degenerateBson))), "\n";' . "\n";
         $expect .= $expectedCanonicalBson . "\n";
 
+        $code .= "\n// Degenerate BSON -> Native -> Canonical BSON\n";
```

Comment:
I believe "Native" in these tests refers to the BSON being deserialised into language native structures (through using `toPHP`. Instead of using "Native" we could use "PHP object" to distinguish it from the tests that decode BSON bytes into a BSON object.